### ISSUE TITLE
Regression in 0.0.62: single-column SQLModel select infers `Sequence[Unknown]` instead of `Sequence[str]`

### ISSUE THREAD

**Issue by jayharrell-tsc** [astral-sh/ty#4094]
### Summary

Starting in `ty` 0.0.62, a single-column SQLModel select is no longer inferred as a scalar sequence. `session.exec(select(Model.str_col)).all()` infers as `Sequence[Unknown]` instead of `Sequence[str]`, so ordinary uses of the result — assigning to `set[str]`, passing to `sorted()` — are reported as errors on code that is correct at runtime.

`ty` 0.0.61 reports `All checks passed!` on the same file. 0.0.62 and 0.0.63 both report 2 errors.

In SQLModel, `select()` on a single column returns `SelectOfScalar[T]`, and `Session.exec()` on a `SelectOfScalar[T]` returns `ScalarResult[T]` — so `.all()` is `Sequence[T]`, not a sequence of rows. The repro below prints the runtime types to confirm the elements really are `str`.

This looks related to #930 (same `session.exec` overload resolution, different overload — that one was `delete()`), but it reproduces on current `main`-era releases and is a regression rather than a never-worked case.

### Reproducer

```python
# /// script
# requires-python = ">=3.13"
# dependencies = ["sqlmodel"]
# ///
from sqlmodel import Field, Session, SQLModel, create_engine, select


class Hero(SQLModel, table=True):
    id: int | None = Field(default=None, primary_key=True)
    name: str


def main() -> None:
    engine = create_engine("sqlite://")
    SQLModel.metadata.create_all(engine)
    with Session(engine) as session:
        session.add(Hero(name="Deadpond"))
        session.commit()

        # A single-column select is a SelectOfScalar[str]; exec() yields str,
        # so .all() is Sequence[str] at runtime.
        names: set[str] = set(session.exec(select(Hero.name)).all())
        ordered: list[str] = sorted(session.exec(select(Hero.name)).all())
        print(names, ordered)


if __name__ == "__main__":
    main()
```

Running it succeeds and prints `{'Deadpond'} ['Deadpond']` — both the `set[str]` and the `sorted()` call behave as annotated.

### Actual behaviour

`ty check repro.py` with `ty==0.0.63`:

```
error[invalid-assignment]: Object of type `set[Sequence[Unknown]]` is not assignable to `set[str]`
  --> repro.py:22:16
   |
22 |         names: set[str] = set(session.exec(select(Hero.name)).all())
   |                --------   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Incompatible value of type `set[Sequence[Unknown]]`
   |                |
   |                Declared type
   |

error[invalid-argument-type]: Argument to function `sorted` is incorrect
  --> repro.py:23:37
   |
23 |         ordered: list[str] = sorted(session.exec(select(Hero.name)).all())
   |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Argument type `Sequence[Unknown]` does not satisfy upper bound `SupportsDunderLT[Any] | SupportsDunderGT[Any]` of type variable `SupportsRichComparisonT`
   |
info: Type variable defined here
   --> stdlib/_typeshed/__init__.pyi:107:1
    |
107 | SupportsRichComparisonT = TypeVar("SupportsRichComparisonT", bound=SupportsRichComparison)  # noqa: Y001
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |

Found 2 diagnostics
```

### Expected behaviour

`All checks passed!`, as on 0.0.61.

### Version bisect

Same file, same `sqlmodel==0.0.39`, one venv per `ty` version:

| `ty` | Result |
|------|--------|
| 0.0.61 | `All checks passed!` |
| 0.0.62 | 2 errors (as above) |
| 0.0.63 | 2 errors (as above) |

So the change landed in 0.0.62.

### Environment

- `ty` 0.0.61 / 0.0.62 / 0.0.63
- `sqlmodel` 0.0.39, `sqlalchemy` 2.0.51
- Python 3.13.13
- macOS, arm64

### Impact

In a real codebase this pattern is common enough to be disruptive — it produced 7 errors across 6 files for us on the 0.0.61 → 0.0.63 bump, all on code verified correct at runtime. The same upgrade also emitted 10 `unused-ignore-comment` warnings for `# ty: ignore` directives that 0.0.61 needed and 0.0.63 does not, so suppressing in place trades one set of diagnostics for another on the next bump. We've pinned `ty>=0.0.61,<0.0.62` for now rather than annotate around it, since the annotations describe the runtime correctly.

**Comment by jayharrell-tsc** [comment]
Cross-checking against other type checkers, since #930 turned out to be a SQLModel typing problem rather than a `ty` one — that isn't the case here. On the same reproducer with the same `sqlmodel==0.0.39`:

| Checker | Result |
|---|---|
| mypy 2.3.0 | `Success: no issues found in 1 source file` |
| mypy 2.3.0 `--strict` | `Success: no issues found in 1 source file` |
| pyright 1.1.411 | `0 errors, 0 warnings, 0 informations` |
| `ty` 0.0.61 | `All checks passed!` |
| `ty` 0.0.62 | 2 errors |
| `ty` 0.0.63 | 2 errors |

So both other checkers resolve the `SelectOfScalar` overload and accept `set[str]` / `sorted(...)`, and `ty` itself did too up to 0.0.61 with SQLModel unchanged. That points at the 0.0.62 change rather than at SQLModel's annotations.

For completeness on the #930 comparison: the SQLModel issue it pointed to (fastapi/sqlmodel#909) was about adding an `exec` overload for `update`/`delete` statements. Single-column `select()` is a different path — SQLModel declares `SelectOfScalar[T]` for it, and `Session.exec()` has an overload returning `ScalarResult[T]`, which is what mypy and pyright follow to `Sequence[str]`.

**Comment by carljm** [comment]
@charliermarsh https://github.com/astral-sh/ruff/pull/27057 fixes this

**Comment by carljm** [comment]
Thanks @jayharrell-tsc for the report!
